graylog2-web-interface/src/contexts/HotkeysModal.tsx

```diff
@@ -0,0 +1,137 @@
+  display: grid;
+  grid-template-columns: auto auto;
+  grid-template-rows: auto;
+  align-items: center;
+  gap: 10px;
+`;
+
+const Content = styled.div`
+  padding: 20px;
+`;
+
+const keyMapper = (key: string) => {
+  const keyMap = {
+    mod: 'command',
+  };
+
+  return keyMap[key] || key;
+};
+
+const Key = ({ description, keys, combinationKey, isEnabled }: { description: string, keys: string, combinationKey: string, isEnabled: boolean}) => {
+  const keysArray = keys.split(combinationKey);
+
+  return (
+    <>
+      <b>{description}:</b>
+      <StyledKeysList>{keysArray.map((key, index) => {
+        const isLast = index === keysArray.length - 1;
+
+        return (
+          <>
+            <Label bsStyle={isEnabled ? 'info' : 'default'}>{keyMapper(key)}</Label>
+            {!isLast && <span>+</span>}
+          </>
+        );
+      })}
+      </StyledKeysList>
+    </>
+  );
+};
+
+const HotkeyCollectionSection = ({ collection, scope }: { collection: HotkeyCollection, scope: ScopeName }) => {
+  const { title, description, actions } = collection;
+  const { activeHotkeys } = useHotkeysContext();
+  const filtratedActions = Object.entries(actions).filter(([actionKey]) => activeHotkeys.has(`${scope}.${actionKey}`),
+    // && (activeHotkeys.get(`${scope}.${actionKey}`)?.options?.enabled !== false);
+  );
+  if (!filtratedActions.length) return null;
+
+  return (
+    <SectionComponent title={title}>
+      <p className="description">{description}</p>
+      <StyledGrid>
+        {
+        filtratedActions.map(([actionKey, { description: keyDescription, keys }]) => {
+          const isEnabled = (activeHotkeys.get(`${scope}.${actionKey}`)?.options?.enabled !== false);
+
+          return (
+            <Key description={keyDescription}
+                 keys={keys}
+                 combinationKey="+"
+                 isEnabled={isEnabled} />
+          );
+        })
+      }
+      </StyledGrid>
+    </SectionComponent>
+  );
+};
+
+const HotkeysModal = () => {
+  const [show, setShow] = useState(false);
+
+  const onHide = useCallback(() => setShow(false), []);
+  const onToggle = useCallback(() => setShow((cur) => !cur), []);
+  const { hotKeysCollection } = useHotkeysContext();
+  useHotkeys('SHOW_HELPER', onToggle, { scopes: 'view' });
+
+  return show && (
+    <Modal show={show}
+           onHide={onHide}
+           bsSize="large">
+      <Modal.Header closeButton>
+        <Modal.Title>Hotkeys</Modal.Title>
+      </Modal.Header>
+
+      <Modal.Body>
+        <Content>
+          <SectionGrid>
+            {
+          Object.entries(hotKeysCollection).map(([scope, collection]: [ScopeName, HotkeyCollection]) => (
+            <HotkeyCollectionSection scope={scope} collection={collection} />
+          ))
+          }
+          </SectionGrid>
+        </Content>
+      </Modal.Body>
+      <Modal.Footer>
+        <LinkButton bsStyle="primary" to="/" target="_blank">Show all hot keys</LinkButton>
```

Comment:
Currently this button is in the position where usually the "Save" modal button is. We could also display it nest to the "general hotkeys" section.

Reply:
Now I saw that `scopes` is part of the `react-hotkeys-hook` options argument.